packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3454,10 +3479,35 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
   @override
   void didUpdateWidget(Navigator oldWidget) {
     super.didUpdateWidget(oldWidget);
-    assert(
-      widget.pages.isEmpty || widget.onPopPage != null,
-      'The Navigator.onPopPage must be provided to use the Navigator.pages API',
-    );
+    assert((){
+      if (widget.pages != const <Page<dynamic>>[]) {
```

Comment:
Is there a particular reason for this condition? Just wondering since that's the only logical difference to the code before and after the change

Reply:
pages by default is `const <Page<dynamic>>[]`, this is make sure user does pass in their own list. This however do not catch the case where user passes in `const <Page<dynamic>>[]`, but this is the best we can do.